src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridEventExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Messaging.EventGrid;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+	/// <summary>
+	/// Extension of the EventGridEvent class for creating <see cref="PushNotification"/> object from an <see cref="EventGridEvent "/> object.
+	/// </summary>
+	public static class EventGridEventExtensions
+    {
+		private const string SyncTokenPropertyName = "syncToken";
+
+		/// <summary>
+		/// This method uses an EventGridEvent from EventGrid and tries to create a <see cref="PushNotification"/>
+		/// </summary>
+		/// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+		/// <param name="pushNotification"> out parameter set up in this method</param>
+		/// <returns></returns>
+		public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+		{
+			pushNotification = null;
+
+			if (eventGridEvent == null || eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
+			{
+				return false;
+			}
+
+			try
+			{
+				string syncToken = (JsonDocument.Parse(eventGridEvent.Data.ToString()).RootElement)
+					.GetProperty(SyncTokenPropertyName).GetString();
```

Comment:
Can this return null?

Reply:
We're catching JsonException, so it looks like we're not trusting the JSON payload completely.

Reply:
Is it possible for EventGridEvent.Data to be invalid JSON?

Reply:
I dont think there's any validation on what's inside `EventGridEvent.Data`. Users may create their own EventGridEvent object and put whatever they want in `Data`: [code](https://github.com/Azure/azure-sdk-for-net/blob/1d8f6fe0207c8f78d10fa9e8e71d823ff751e88f/sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridEvent.cs#L52)

Reply:
In that case, need to check if the property exists on the JsonDocument